lang/LangSource/PyrInterpreter3.cpp

```diff
@@ -1652,31 +1652,45 @@ HOT void Interpret(VMGlobals *g)
 				// Integer-forBy : 143 7, 143 8, 143 9
 				case 7 : {
 					PyrSlot * vars = g->frame->vars;
-					if (IsFloat(vars+1)) {
-						SetInt(&vars[1], (int32)(slotRawFloat(&vars[1])));
-					}
-					if (IsFloat(vars+2)) {
-						SetInt(&vars[2], (int32)(slotRawFloat(&vars[2])));
-					}
-					int tag = GetTag(&vars[1]);
-					if ((tag != tagInt)
-							|| NotInt(&vars[2])) {
-						error("Integer-forBy : endval or stepval not an Integer.\n");
+					// If any argument is floating point we cast all arguments
+					// to floating point, including the accumulator. This avoids
+					// potential infinite loops due to integer truncation.
+					if (IsFloat(vars+1) || IsFloat(vars+2)) {
+						if (IsInt(vars+1)) {
+							SetFloat(&vars[1], (double)(slotRawInt(&vars[1])));
+						}
+						if (IsInt(vars+2)) {
+							SetFloat(&vars[2], (double)(slotRawInt(&vars[2])));
+						}
+						SetFloat(&vars[4], (double)(slotRawInt(&g->receiver)));
+					} else  {
+						int tag = GetTag(&vars[1]);
+						if ((tag != tagInt)
+								|| NotInt(&vars[2])) {
+							error("Integer-forBy : endval or stepval not an Integer.\n");
 
-						slotCopy(++sp, &g->receiver);
-						numArgsPushed = 1;
-						selector = gSpecialSelectors[opmPrimitiveFailed];
-						slot = sp;
+							slotCopy(++sp, &g->receiver);
+							numArgsPushed = 1;
+							selector = gSpecialSelectors[opmPrimitiveFailed];
+							slot = sp;
 
-						goto class_lookup;
+							goto class_lookup;
+						}
+						slotCopy(&vars[4], &g->receiver);
 					}
-					slotCopy(&vars[4], &g->receiver);
 					dispatch_opcode;
 				}
 				case 8 : {
 					PyrSlot * vars = g->frame->vars;
-					if ((slotRawInt(&vars[2]) >= 0 && slotRawInt(&vars[4]) <= slotRawInt(&vars[1]))
-							|| (slotRawInt(&vars[2]) < 0 && slotRawInt(&vars[4]) >= slotRawInt(&vars[1]))) {
+					bool continue_for_by = false;
```

Comment:
the codebase isn't super consistent but I'd say we generally prefer camelCase naming for local variables.

Reply:
Done.